org.lflang/src/org/lflang/util/LFCommand.java

```diff
@@ -0,0 +1,312 @@
+     */
+    protected LFCommand(ProcessBuilder pb) { processBuilder = pb; }
+
+
+    /**
+     * Get the output collected during command execution
+     */
+    public OutputStream getOutput() { return output; }
+
+
+    /**
+     * Get the error output collected during command execution
+     */
+    public OutputStream getErrors() { return errors; }
+
+
+    /**
+     * Get a String representation of the stored command
+     */
+    public String toString() { return String.join(" ", processBuilder.command()); }
+
+
+    /**
+     * A runnable that collects the output from a running process, prints it and stores it in the output stream
+     */
+    private class OutputCollector implements Runnable {
+
+        private final Process process;
+
+
+        OutputCollector(Process process) {
+            this.process = process;
+        }
+
+
+        @Override
+        public void run() {
+            byte[] buffer = new byte[64];
+            int len;
+            do {
+                try {
+                    len = process.getInputStream().read(buffer);
+                    if (len > 0) {
+                        output.write(buffer, 0, len);
+                        System.out.write(buffer, 0, len);
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    break;
+                }
+            } while (len != -1);
+        }
+    }
+
+    /**
+     * A runnable that collects the error output from a running process, prints it and stores it in the output stream
+     */
+    private class ErrorCollector implements Runnable {
+
+        private final Process process;
+
+
+        ErrorCollector(Process process) {
+            this.process = process;
+        }
+
+
+        @Override
+        public void run() {
+            byte[] buffer = new byte[64];
+            int len;
+            do {
+                try {
+                    len = process.getErrorStream().read(buffer);
+                    if (len > 0) {
+                        errors.write(buffer, 0, len);
+                        System.err.write(buffer, 0, len);
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    break;
+                }
+            } while (len != -1);
+        }
+    }
+
+
+    /**
+     * Execute the command while forwarding output and error streams.
+     * <p>
+     * Executing a process directly with `processBuiler.start()` could
+     * lead to a deadlock as the subprocess blocks when output or error
+     * buffers are full. This method ensures that output and error messages
+     * are continuously read and forwards them to the system output and
+     * error streams as well as to the output and error streams hold in
+     * this class.
+     *
+     * @return the process' return code
+     * @author {Christian Menard <christian.menard@tu-dresden.de}
+     */
+    public Integer run() {
```

Comment:
We have very similar code in the `org.lflang.test` package that is used to execute tests and capture their output. We should probably update that code and leverage this class instead...

Reply:
Absolutely! I opened #418 to keep track of this. I was planning to use the new command infrastructure for implementing #320 anyway, and we can update the other tests in the same go.